### PR TITLE
chore(db): Migrate CreateApplicationVersion to export-service: apps table

### PR DESCRIPTION
**Pull request by sven-urbanski-freiheit-com**
This migrates the "app" part of the CreateApplicationVersion-transformer to the export-service.
The "releases" table will be done in a separate PR.

This also removes the "SourceRepoUrl" parameter, as it was not working to begin with.

Ref: SRX-L070MM

### CODE REVIEW

File: pkg/db/db.go [hunk 1]
```diff
@@ -751,6 +777,163 @@ func (h *DBHandler) DBSelectAnyDeployment(ctx context.Context, tx *sql.Tx) (*DBD
...
+	selectQuery := h.AdaptQuery(fmt.Sprintf(
+		"SELECT eslVersion, appName, metadata " +
+			" FROM apps " +
+			" LIMIT 1;"))
+	rows, err := tx.QueryContext(
+		ctx,
+		selectQuery,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("could not query esl table from DB. Error: %w\n", err)
```
Comment (miguel-crespo-fdc) [inline pkg/db/db.go:843]: apps table

Comment (miguel-crespo-fdc) [reply pkg/db/db.go:843]:
```suggestion
		return nil, fmt.Errorf("could not query apps table from DB. Error: %w\n", err)
```

File: services/cd-service/pkg/repository/transformer.go [hunk 2]
```diff
@@ -1767,17 +1767,27 @@ func (c *CreateEnvironmentTeamLock) Transform(
...
+				logger.FromContext(ctx).Sugar().Warnf("CreateEnvironmentTeamLock: Could not find team for application: %s.", currentApp)
+			} else {
+				if c.Team == currentTeamName {
 					foundTeam = true
 					break
 				}
-			} else {
-				logger.FromContext(ctx).Sugar().Warnf("CreateEnvironmentTeamLock: Could not find team for application: %s.", currentApp)
 			}
+			//currentTeamFile := fs.Join("applications", currentApp, "team")
```
Comment (miguel-crespo-fdc) [inline services/cd-service/pkg/repository/transformer.go:1782]: Commented code

File: pkg/db/db.go [hunk 3]
```diff
@@ -751,6 +777,163 @@ func (h *DBHandler) DBSelectAnyDeployment(ctx context.Context, tx *sql.Tx) (*DBD
...
+type DBApp struct {
+	EslId EslId
+	App   string
+}
+
+type DBAppMetaData struct {
+	Team string
+}
+
+type DBAppWithMetaData struct {
```
[the remark this answers is not in the document]
Comment (sven-urbanski-freiheit-com) [reply pkg/db/db.go:789]: In the src, not really, no. But I will add a test.

[the remark this answers is not in the document]
Comment (sven-urbanski-freiheit-com) [reply pkg/db/db.go:786]: done

**Review by miguel-crespo-fdc — APPROVED** [review]

File: pkg/db/db.go [hunk 4]
```diff
@@ -1579,6 +1579,14 @@ func (h *DBHandler) DBDeleteEnvironmentLock(ctx context.Context, tx *sql.Tx, env
 
 	if existingEnvLock == nil {
 		previousVersion = 0
+		existingEnvLock = &EnvironmentLock{
```
[the remark this answers is not in the document]
Comment (sven-urbanski-freiheit-com) [reply pkg/db/db.go]: done (I cherry-picked your PR #1658 )

**Pull request merged**